common/app/model/RelatedContent.scala

```diff
@@ -39,13 +39,19 @@ object RelatedContent {
 
 object StoryPackages {
   def apply(parent: ContentType, response: ItemResponse): RelatedContent = {
-    val storyPackagesContent = response.packages.map { packages =>
-      packages.map { p =>
-        p.articles.map(_.content)
+
+    val storyPackagesContent: Seq[ApiContent] = response.packages.map { packages =>
+      val allContentsPerPackage: Seq[Seq[ApiContent]] = packages.map(_.articles.map(_.content))
+      if(packages.size > 1) { //intermix packages only if more than one
+        allContentsPerPackage
+         .flatMap(_.zipWithIndex) // zip content with its position
+         .sortBy(_._2).map(_._1)  // sort by position and extract content
+         .groupBy(_.id).map(_._2.head).toList // remove duplicates (not using `distinct` here that would require
```

Comment:
I feel like doing groupBy since it produces a map, will lose the ordering created in the previous step